codebasin/preprocessor.py

```diff
@@ -619,35 +605,19 @@ def visit(self, visitor: Callable[[Self], Visit]):
                 child.visit(visitor)
 
 
+@dataclass(eq=False)
 class FileNode(Node):
     """
     Typically the root node of a tree. Simply contains a filename after
     inheriting from the Node class.
     """
 
-    def __init__(self, _filename):
-        super().__init__()
-        self.filename = _filename
-        # The length of the file, counting blank lines and comments
-        self.num_lines = 0
-        # The source lines of code, ignoring blank lines and comments
-        self.total_sloc = 0
-        self.file_hash = self.__compute_file_hash()
-
-    def __compute_file_hash(self):
-        chunk_size = 4096
-        hasher = hashlib.sha512()
-        with open(self.filename, "rb") as in_file:
-            for chunk in iter(lambda: in_file.read(chunk_size), b""):
-                hasher.update(chunk)
-
-        return hasher.hexdigest()
-
-    def __repr__(self):
-        return _representation_string(self, attrs=["filename"])
+    filename: str
+    num_lines: int = field(default=0, init=False)
+    total_sloc: int = field(default=0, init=False)
```

Comment:
Same comment as in `Node`: unless I'm missing a reason for it, the only instances where you **have** to use `field` is for empty lists and `dict`s. If you're just going to have zero as the default, `int = 0` should suffice.

I guess it just makes keeping a consistent style easier?